dev/ColorPicker/readme.md

```diff
@@ -0,0 +1,141 @@
+
+The `ColorSpectrum` also maintains a cursor that visually identifies to the user the selected color. When the spectrum image is rendered (more on that below) an array of the color at every pixel in the spectrum image is stored (`m_hsvValues`). This array can become quite large, for example a 600px x 600px spectrum will build an array of 36000 unique HSV colors. When the pointer is pressed on the spectrum, the corresponding color at those X/Y pointer coordinates is looked-up in the array and set as the new selected color. The look-up is simple as the array indexes directly correspond to the pixel X/Y coordinates.
+
+Note: Because the `ColorSpectrum` works in HSV, it is very important that the `HsvColor` property is used to get/set the selected color on the spectrum instead of `Color` (which uses RGB). This is necessary to avoid the accumulation of rounding errors when converting between HSV/RGB. This also means that the `HsvColor` must be used in code-behind instead of binding to `Color` in XAML. Remember the HSV color is the main color representation internally -- not RGB.
+
+### Template Parts
+
+The color spectrum looks for and uses the following parts in the control template. 
+
+* `LayoutRoot` : [Grid] The root element of the control. Changes to the size of the root element will trigger re-calculation of the spectrum.
+* `SizingGrid` : [Grid] The grid representing the actual width/height that the spectrum will be drawn to.
+* `SpectrumRectangle` : [Rectangle] The background rectangle use to represent the color spectrum gradient. This is blended with the overlay rectangle using opacity (see notes below on rendering). This rectangle is only visible in Box shape.
+* `SpectrumEllipse` : [Ellipse] The background ellipse use to represent the color spectrum gradient. This is blended with the overlay ellipse using opacity (see notes below on rendering). This ellipse is only visible in Ring shape.
+* `SpectrumOverlayRectangle` : [Rectangle] The overlay rectangle use to represent the color spectrum gradient. This is blended with the background rectangle using opacity (see notes below on rendering). This rectangle is only visible in Box shape.
+* `SpectrumOverlayEllipse` : [Ellipse] The overlay ellipse use to represent the color spectrum gradient. This is blended with the background ellipse using opacity (see notes below on rendering). This ellipse is only visible in Ring shape.
+* `InputTarget` : [FrameworkElement] The control used to track pointer interaction by the user and get X/Y coordinates to change the selected color in the spectrum.
+* `SelectionEllipsePanel` : [Panel] The panel representing the selected color indicator shown on the spectrum. 
+* `ColorNameToolTip` : [ToolTip] The tool-tip used to present the selected color's displayed name to the user.
+
+### Rendering the Spectrum
+
+There are presently two different techniques used for generating/rendering the color spectrum depending on the version of Windows.
```

Comment:
nit: remove presently

Reply:
I can remove that. In my mind I know this is going to change in the future which is why I think it's better to qualify it with a time. In WinUI 3.0 for example I'm sure the pre-RS2 method of rendering will go away.

Reply:
Well a lot of these sections will change over time. Having presently does not help if it's outdated, instead we need to watch out that the documentation stays up to date.

Reply:
It was removed either way